File: spark/src/test/scala/geotrellis/spark/io/avro/TileRecordSpec.scala
package geotrellis.spark.io.avro

import geotrellis.spark.GridKey
import geotrellis.spark.io.avro.codecs.{KeyValueRecordCodec, TileCodecs, KeyCodecs}
import TileCodecs._
import KeyCodecs._
import org.scalatest._
import geotrellis.raster._

class TileRecordSpec extends FunSpec with AvroTools with Matchers {
  describe("TileRecordCodecs") {
    it("encodes (key,tile) pairs"){
      val pairs = Vector(
        GridKey(1,2) -> DoubleArrayTile.fill(1,10,12),
        GridKey(3,6) -> DoubleArrayTile.fill(2,10,12),
        GridKey(3,4) -> DoubleArrayTile.fill(3,10,12)
      )

      val codec = new KeyValueRecordCodec[GridKey, DoubleArrayTile]

      roundTrip(pairs)(codec)
    }
  }
}

File: geotrellis-python/tests/geotrellis/spark/io/avro/TileRecordSpec.py
from tests.geotrellis.spark.io.avro.AvroTools import AvroTools
from geotrellis.spark.io.avro.codecs.KeyValueRecordCodec import KeyValueRecordCodec
from geotrellis.raster.DoubleArrayTile import DoubleArrayTile
from geotrellis.spark.SpatialKey import SpatialKey
from spec import Spec

class TileRecordSpec(AvroTools, Spec):
    "TileRecordCodecs"

    def test_pairs(self):
        "encodes (key,tile) pairs"

        pairs = [(SpatialKey(1,2), DoubleArrayTile.fill(1,10,12)),
                (SpatialKey(3,6), DoubleArrayTile.fill(2,10,12)),
                (SpatialKey(3,4), DoubleArrayTile.fill(3,10,12))]

        codec = KeyValueRecordCodec(SpatialKey, DoubleArrayTile)

        self.roundTrip(pairs, codec)
